tests/ui/moves/recreating-value-in-loop-condition.rs

```diff
@@ -0,0 +1,57 @@
+fn iter<T>(vec: Vec<T>) -> impl Iterator<Item = T> {
+    vec.into_iter()
+}
+fn foo() {
+    let vec = vec!["one", "two", "three"];
+    while let Some(item) = iter(vec).next() { //~ ERROR use of moved value
+        println!("{:?}", item);
+    }
+}
```

Comment:
Could you add the appropriate `//~ HELP` lines as well? You can probably inline `iter()` to avoid the "consider changing this parameter to borrow" suggestion

Reply:
> You can probably inline iter() to avoid the "consider changing this parameter to borrow" suggestion

If I do that the error *doesn't* happen, with a logic error because then the iterator gets recreated on every loop iteration, turning it into an infinite loop.

Reply:
The error does happen on `while let Some(item) = vec.into_iter().next()`, is that not what we're testing?

Reply:
Ah, I was worried you'd have to add a bunch of `NOTE`s because of `iter()` but apparently not so all's good